src/test/java/com/exadel/discount/service/CountryServiceTest.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+public class CountryServiceTest extends AbstractIT {
+
+    private static final UUID ID = UUID.fromString("971bf698-f3ea-4a97-85e8-0a2a770736d6");
+
+    @Autowired
+    private CountryServiceImpl countryService;
+
+    @Mock
+    private CountryRepository countryRepository;
+
+    @Mock
+    private CountryMapper mapper;
+
+    private CountryDTO input;
+
+    @BeforeEach
+    public void setUp() {
+        input = CountryDTO.builder()
+                .name("test")
+                .build();
+    }
+
+
+    @Test
+    public void testSave() {
+        var result = countryService.save(input);
+        Assertions.assertNotNull(result);
+    }
+
+    @Test
+    public void testFindALl() {
+        List<Country> expected = List.of(new Country(), new Country());
+
+        when(countryRepository.findAll()).thenReturn(expected);
+        CountryServiceImpl implS = Mockito.mock(CountryServiceImpl.class);
+        List<CountryDTO> actual = implS.findAll();
+
+        Assertions.assertEquals(actual, mapper.getListDTO(expected));
+    }
+
+
+    public void testFindById() {
+        Country country = mapper.countryDTOToCountry(input);
+        when(countryRepository.findById(UUID.randomUUID())).thenReturn(Optional.of(country));
+        CountryDTO actual = countryService.findById(UUID.randomUUID());
+
+        Assertions.assertEquals(input,actual);
+    }
+
+    @Test
+    public void testExceptionFindById(){
+        Exception exception = assertThrows(NotFoundException.class,
+                ()-> countryService.findById(ID));
+        Assertions.assertEquals("Country with id "+ID+" not found",exception.getMessage());
+
+    }
+
+    @Test
+    public void testDeleteById(){
+        when(countryRepository.findById(ID)).thenReturn(Optional.of(new Country()));
+        doNothing().when(countryRepository).delete(any(Country.class));
+        countryRepository.deleteById(UUID.randomUUID());
+    }
+
+    @Test
+    public void testExceptionDeleteById(){
+        Exception exception = assertThrows(NotFoundException.class,
+                () -> countryService.deleteById(ID));
+        Assertions.assertEquals("Country with Id "+ID+" not found",exception.getMessage());
+    }
+
+
+    public void testFindByName(){
+
+        Country country = mapper.countryDTOToCountry(input);
+        when(countryRepository.findByName(anyString())).thenReturn(Optional.of(country));
+        CountryServiceImpl c = Mockito.mock(CountryServiceImpl.class);
+        CountryDTO actual = c.findByName(anyString());
+
+        Assertions.assertEquals(actual,input);
+
+    }
+
+    @Test
+    public void testExceptionFindByName(){
```

Comment:
rename to 
testFindByNameThrowsNotFoundException

Reply:
done